v2/devdoc/log_context_requirements.md

```diff
@@ -0,0 +1,152 @@
+
+`log_context` encapsulates storing a series of properties and their values (that meant to be used when logging is done).
+
+A `log_context` holds a count of properties and the backing memory to hold the properties name and values.
+
+## Exposed API
+
+```
+typedef struct LOG_CONTEXT_TAG* LOG_CONTEXT_HANDLE;
+
+uint32_t log_context_get_property_value_pair_count(LOG_CONTEXT_HANDLE log_context);
+const LOG_CONTEXT_PROPERTY_VALUE_PAIR* log_context_get_property_value_pairs(LOG_CONTEXT_HANDLE log_context);
+
+#define LOG_CONTEXT_CREATE(dest_log_context, parent_context, ...) \
+    ...
+
+#define LOG_CONTEXT_DESTROY(log_context) \
+    ...
+
+#define LOG_MAX_STACK_DATA_SIZE                 1024
+#define LOG_MAX_STACK_PROPERTY_VALUE_PAIR_COUNT 64
+```
+
+## LOG_CONTEXT_CREATE
+
+```c
+#define LOG_CONTEXT_CREATE(dest_log_context, parent_context, ...) \
+    ...
+```
+
+`LOG_CONTEXT_CREATE` creates a dynamically allocated log context.
+
+`...` is a list of `LOG_CONTEXT_PROPERTY`, `LOG_CONTEXT_STRING_PROPERTY` or `LOG_CONTEXT_NAME` entries.
+
+**SRS_LOG_CONTEXT_01_001: [** `LOG_CONTEXT_CREATE` shall allocate memory for the log context. **]**
+
+**SRS_LOG_CONTEXT_01_013: [** `LOG_CONTEXT_CREATE` shall store one property/value pair that with a property type of `struct` with as many fields as the total number of properties passed to `LOG_CONTEXT_CREATE`. **]**
+
+**SRS_LOG_CONTEXT_01_012: [** The name of the `struct` property shall be the context name specified by using `LOG_CONTEXT_NAME` (if specified). **]**
+
+**SRS_LOG_CONTEXT_01_003: [** `LOG_CONTEXT_CREATE` shall store the property types and values specified by using `LOG_CONTEXT_PROPERTY` in the context. **]**
+
+**SRS_LOG_CONTEXT_01_014: [** If `parent_context` is non-`NULL`, the created context shall include all the property/value pairs of `parent_context`. **]**
+
+**SRS_LOG_CONTEXT_01_002: [** If any error occurs, `LOG_CONTEXT_CREATE` shall fail and return `NULL`. **]**
+
+## LOG_CONTEXT_DESTROY
+
+```c
+#define LOG_CONTEXT_DESTROY(log_context) \
+    ...
+```
+
+`LOG_CONTEXT_DESTROY` frees up all resources associated with `log_context`.
+
+**SRS_LOG_CONTEXT_01_006: [** `LOG_CONTEXT_DESTROY` shall free the memory and resources associated with `log_context` that were allocated by `LOG_CONTEXT_CREATE`. **]**
+
+## LOG_CONTEXT_PROPERTY
+
+```c
+#define LOG_CONTEXT_PROPERTY(property_type, property_name, property_value) \
+    ...
+```
+
+`LOG_CONTEXT_PROPERTY` allows defining a property/value pair of a given supported type.
+
+**SRS_LOG_CONTEXT_01_004: [** `LOG_CONTEXT_PROPERTY` shall expand to code allocating a property/value pair entry with the type `property_type` and the name `property_name`. **]**
+
+**SRS_LOG_CONTEXT_01_005: [** `LOG_CONTEXT_PROPERTY` shall expand to code copying the value `property_value` to be the value of the property/value pair. **]**
+
+**SRS_LOG_CONTEXT_01_019: [** If 2 properties have the same name for a context a compiler error shall be emitted. **]**
```

Comment:
>name for a context [](http://example.com/codeflow?start=60&length=18)

have the same "property_name" a compiler...

Reply:
Updated.